fiber/chain_interactions/commitments.py

```diff
@@ -0,0 +1,224 @@
+from enum import Enum
+from typing import TypeAlias
+
+from pydantic import BaseModel
+from scalecodec import ScaleType
+from substrateinterface import SubstrateInterface, Keypair
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from fiber.chain_interactions.chain_utils import format_error_message
+
+
+class DataFieldType(Enum):
+    RAW = "Raw"
+    BLAKE_TWO_256 = "BlakeTwo256"
+    SHA_256 = "Sha256"
+    KECCAK_256 = "Keccak256"
+    SHA_THREE_256 = "ShaThree256"
+
+
+CommitmentDataField: TypeAlias = tuple[DataFieldType, bytes] | None
+
+
+class CommitmentQuery(BaseModel):
```

Comment:
in /chain_interactions (soon to be renamed /chain) there is a models.py where these could live